packages/block-library/src/comments-query-loop/edit/toolbar.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	ToolbarGroup,
+	Dropdown,
+	ToolbarButton,
+	BaseControl,
+	__experimentalNumberControl as NumberControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { settings } from '@wordpress/icons';
+
+export default function CommentsQueryLoopToolbar( {
+	attributes,
+	setAttributes,
+} ) {
+	const { perPage } = attributes;
+	return (
+		<ToolbarGroup>
+			<Dropdown
+				renderToggle={ ( { onToggle } ) => (
+					<ToolbarButton
+						icon={ settings }
+						label={ __( 'Discussion Settings' ) }
+						onClick={ onToggle }
+					/>
+				) }
+				renderContent={ () => (
+					<>
+						<BaseControl>
+							<NumberControl
+								__unstableInputWidth="60px"
```

Comment:
>This is what we see on the toolbar right now. Is there any way to make it wider without custom CSS?

You'll have to change this value here.

Also in order to have a wider popover you can add an extra css class. At least this is what we're doing for `Query Loop`: https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query/edit/query-toolbar.js#L47